src/tensorflow/lite/micro/tools/project_generation/create_tflm_arduino.py

```diff
@@ -0,0 +1,548 @@
+import configparser
+
+# parse args: {outdir}
+# generate list of all source/header files and their src/dst locations
+# create directory tree in {outdir}/tflm_arduino
+# download third-party source
+# copy templates: library.properties to {outdir}/tflm_arduino
+# copy templates: TensorFlowLite.h to {outdir}/tflm_arduino/src
+# copy transforms: examples
+#   --platform=arduino
+#   --third_party_headers=<output from make
+#       list_{example_name}_example_headers
+#       list_third_party_headers>
+#   --is_example_source
+# copy transforms: examples/{example_name}/main_functions.cc to
+#  {example_name}.ino
+#   --platform=arduino
+#   --third_party_headers=<output from make
+#       list_{example_name}_example_headers
+#       list_third_party_headers>
+#   --is_example_ino
+# copy transforms: tensorflow, third_party
+#   --platform=arduino
+#   --third_party_headers=<output from make
+#       list_third_party_headers>
+# patch third_party/flatbuffers/include/flatbuffers/base.h with:
+#   sed -E 's/utility\.h/utility/g'
+# patch third_party/kissfft/kiss_fft.h
+#   sed -E 's@#include <string.h>@//#include <string.h>
+# 	 /* Patched by helper_functions.inc for Arduino compatibility */@g'
+# run fix_arduino_subfolders.py {outdir}/tflm_arduino
+# remove all empty directories in {outdir}/tflm_arduino tree
+# create ZIP file using shutil.make_archive()
+
+
+def RunSedScripts(file_path: Path,
+                  scripts: List[str],
+                  args: Union[str, None] = None,
+                  is_dry_run: bool = True) -> None:
+  """
+  Run SED scripts with specified arguments against the given file.
+  The file is updated in place.
+
+  Args:
+    file_path: The full path to the input file
+    scripts: A list of strings, each containing a single SED script
+    args: a string containing all the SED arguments | None
+    is_dry_run: if True, do not execute any commands
+  """
+  cmd = "sed"
+  if args is not None and args != "":
+    cmd += f" {args}"
+  for scr in scripts:
+    cmd += f" -E {scr}"
+  cmd += f" < {file_path!s}"
+  print(f"Running command: {cmd}")
+  if not is_dry_run:
+    result = subprocess.run(cmd,
+                            shell=True,
+                            check=True,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE)
+    print(f"Saving output to: {file_path!s}")
+    file_path.write_bytes(result.stdout)
+
+
+def RemoveDirectories(paths: List[Path], is_dry_run: bool = True) -> None:
+  """
+  Remove directory tree(s) given list of pathnames
+
+  Args:
+    paths: A list of Path objects
+    is_dry_run: if True, do not execute any commands
+  """
+  for dir_path in paths:
+    print(f"Removing directory tree {dir_path!s}")
+    if dir_path.exists() and not is_dry_run:
+      shutil.rmtree(dir_path)
+
+
+def RemoveEmptyDirectories(paths: Iterable[Path],
+                           root: Path,
+                           is_dry_run: bool = True) -> None:
```

Comment:
not clear to me why in the overal generation process we will have empty directories. Let's talk about the need for this function.